mod/state-transition/pkg/core/deposits_validation.go

```
@@ -0,0 +1,160 @@
+
+		if len(localDeposits) != len(deposits) {
+			return errors.Wrapf(
+				ErrDepositsLengthMismatch,
+				"local: %d, payload: %d", len(localDeposits), len(deposits),
+			)
+		}
+
+		for i, sd := range localDeposits {
+			// Deposit indices should be contiguous
+			//#nosec:G701 // i never negative
+			expectedIdx := expectedStartIdx + uint64(i)
```

golangci-lint (1.62.2)

[high] G115: integer overflow conversion int -> uint64

(gosec)